app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -610,15 +610,26 @@ export default {
       // reload the queries if the response was successfull
       // 200 OK (PUT) / 201 Created (POST)
       if ([200, 201].includes(status)) {
-        this.isQueryModified = false;
         //Show a message for the user, your query is saved
         this.isQuerySaved = true;
         this.enabledQuerySavedButton = false
         this.isQueryModified = false
         this.isQuerySavingPromptVisible = false
 
-        this.setPublicQueries(await this.getPublicQueries());
         this.setPrivateQueries(await this.getPrivateQueries());
+
+        //Get the old list
+        const oldQueries = this.publicQueries
+        //Get the list with the new query
+        const newQueries = this.privateQueries
```

Comment:
This is confusing. `publicQueries` and `privateQueries` store different stuff, why now is `old` and `new`? Wha's the point?

Reply:
For example, you save a new query, at that moment, that query is not yet in `this.publicQueries`, but it's in `this.privateQueries` since it has been updated a [few lines before](https://github.com/PopulateTools/gobierto/blob/e0b7450a19612fbe6fc688db810d819603db0e99/app/javascript/gobierto_data/webapp/components/Dataset.vue#L619). 

```Javascript
console.log(this.publicQueries.length) 5
console.log(this.privateQueries.length) 6
```

So I can compare the list before saving with the list containing the query I just kept. Then I filter them, and so I get the new saved query.

And now we [updated](https://github.com/PopulateTools/gobierto/blob/e0b7450a19612fbe6fc688db810d819603db0e99/app/javascript/gobierto_data/webapp/components/Dataset.vue#L632) `this.publicQueries` 

```Javascript
console.log(this.publicQueries.length) 6
console.log(this.privateQueries.length) 6
```